### PR TITLE
UCP/TAG/TEST: Fix handling empty IOV buffers

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7086/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7087/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7089/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7091/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7094/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 1]
```diff
@@ -952,6 +952,42 @@ UCS_TEST_P(test_ucp_tag_xfer, test_xfer_len_offset, "RNDV_THRESH=1000") {
...
+    const int    sync      = 0;
+    const int    truncated = 0;
+
+    std::vector<char> sendbuf(size, 0);
+    std::vector<char> recvbuf(size, 0);
+    ucp_dt_iov_t iovec[iovcnt];
+
+    ucs::fill_random(sendbuf);
+
+    /* initialize iovec with 99 empty buffers and one non-empty */
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_tag_xfer.cc]: minor: 39?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7098/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7101/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7106/ for details (Mellanox internal link).

**Pull request merged**